docs: Reshape Server fundamentals + add Configuration reference (#672)
serverpod/serverpod_docs: docs/06-concepts/07-configuration.md

---

> description: "Configure Serverpod with YAML files, environment variables, or a Dart config object: run modes, the three configuration sources and their precedence, and secrets."

Reviewer: Descriptions are unquoted across the docs (123 of 128 pages). Lead with the noun and drop the colon so it needs no quotes.

```suggestion
description: Configuration in Serverpod comes from YAML files, environment variables, or a Dart config object, with run modes, source precedence, and secrets.
```

>   scanInterval: 2000

Reviewer: These aren't the current defaults. `serverpod_shared/lib/src/config.dart` sets `concurrencyLimit` to `1` and `scanInterval` to `5000`, which matches the new reference table. `concurrencyLimit: 5` on the line above should be `1` too (it's outside the diff so I couldn't include it in the suggestion).

```suggestion
  scanInterval: 5000
```

final stripeApiKey = pod.getPassword('stripeApiKey');
```

This works for built-in and custom secrets alike, whether they come from the passwords file or an environment variable.

### Secrets in production

A new project's `.gitignore` excludes `config/passwords.yaml` and credential files such as `config/firebase_service_account_key.json`, so secrets are not committed by default. Keep production secrets out of source control.

In production, set secrets through `SERVERPOD_PASSWORD_*` environment variables, or your host's secret manager, rather than a checked-in passwords file.

### Passwords on Serverpod Cloud

On [Serverpod Cloud](/cloud), the values you read with `getPassword` live in the **Passwords** tier. Set them from the command line instead of editing a passwords file:

```bash
scloud password set stripeApiKey "sk_live_..."
```

Use `--from-file` for long or multi-line values such as a service account JSON. Cloud stores each password encrypted and injects it so `getPassword` reads it exactly as it does locally. See [Passwords, secrets, and environment variables](/cloud/concepts/passwords-secrets-env-vars) for the full reference.

> ### Passwords file example

Reviewer: Remove this section, it's a duplicate. The prose repeats the Secrets intro (line 153) almost verbatim, the YAML is the same example already shown under "Via the passwords file" (180-196), and the SQLite note is already covered in the SQLite section (line 102). Nothing here is lost by dropping it.

Author: Removed.
